tests/ui/macro_use_imports.rs

```diff
@@ -1,11 +1,36 @@
-// edition:2018
+// compile-flags: --edition 2018
+// aux-build:macro_rules.rs
+// aux-build:macro_use_helper.rs
+
+#![allow(clippy::single_component_path_imports)]
 #![warn(clippy::macro_use_imports)]
 
-use std::collections::HashMap;
 #[macro_use]
-use std::prelude;
+extern crate macro_use_helper as mac;
+
+#[macro_use]
+extern crate clippy_mini_macro_test as mini_mac;
+
+mod a {
+    #[macro_use]
+    use mac;
```

Comment:
Can you add a test for multi-macro use from the same module/crate?

Reply:
What would be an example of this, like importing multiple macros from the same mod/crate?

Reply:
```rust
// some_mod.rs
macro_rules! m1 { ... }
macro_rules! m2 { ... }
```

```rust
// main.rs
#[macro_use] 
use some_mod;

fn main() {
    m1!();
    m2!();
}
```

should suggest

```
use some_mod::{m1, m2};
```

Reply:
I have a `HashMap` of module name to an item which works fine as long as there isn't nested modules, any ideas how to handle `my_mod::{nested_mod::{macro_a, macro_b}}`?

Reply:
You can look at the `wildcard_imports` module. I have written code that builds suggestions for this. You can take a look at this and maybe extract it to `utils` and use the suggestion generation in both modules.

Reply:
I was hoping you were going to say something like that what I implemented is shaky at best.

Reply:
Using the wildcard_imports code was problematic, we need to have all the information (the imports and the macro uses) available I don't see a way to deal with each import and list of macros as they are encountered (is there a macro equivalent to the `names_imported_by_glob_use` method?). Is there a way to declare a lint struct with a lifetime (`impl_lint_pass!(MacroUseImports<'item_lifetime> => [MACRO_USE_IMPORTS]);`) then it would be possible to save the `Item` or `Path` needed to do something more like wildcard_imports.

Reply:
> (is there a macro equivalent to the names_imported_by_glob_use method?

nope sadly not. At least I don't know one.

> Is there a way to declare a lint struct with a lifetime

A small rustc change would be necessary. I also encountered that recently. I will open a rustc PR for this.

Reply:
rust-lang/rust#72657

Reply:
@DevinR528 Now you can define lint passes with lifetimes and use `impl_lint_pass`

Reply:
Cool! Thanks for letting me know. I will get back to it sometime in the coming week. Hope that's ok? GSoC started!